### DISCUSSION TITLE
Expose `autoCancelAfterFailures` in Cypress executor

### DISCUSSION THREAD

**Discussion by nsmith7989** [nrwl/nx#26264] [Feature Requests]
[Cypress's module API](https://docs.cypress.io/guides/guides/module-api) exposes an option for controlling the auto cancellation feature on Cypress cloud. In my setup, exposing this option would fix an issue in DTE.

## DTE issue with cypress auto cancellation
DTE mode with atomized cypress tests and cypress cloud with auto cancellation turned on. Say there are 4 specFiles on a project, those 4 specFiles have been assigned their own target via the cypress project crystal plugin. When running specFile 1 fails while the other specFiles are executing. With cypress auto cancellation, as soon as a single spec file fails, it will cancel all other tests. Those other specs will cancel in the middle of a run with a `0` exit status. This could mean that not all of the specs will have run, but NX cloud sees a `0` exit code and marks that task as successful and cached. A re-run means that a partially complete specFile will not re-run, but there may be tests within that specFile that never executed. 

Turning off this cypress cloud feature will fix this issue for us, as it will not cancel the run if X number of tests fail. 

### Can you change this within cypress dashboard directly? 
Yes, this is a setting within cypress. However our setup has both DTE mode mixed with standard mode. In non DTE mode the cypress auto cancelation is useful as it saves time, so we do not want to blanket disable it.

**Comment by nsmith7989** [comment]
Implemented

**Comment by leosvelperez** [comment]
Addressed by https://github.com/nrwl/nx/pull/26281.
